src/Adapter/TaxRulesGroup/CommandHandler/EditTaxRulesGroupHandler.php

```diff
@@ -55,29 +72,17 @@ public function handle(EditTaxRulesGroupCommand $command): void
             $taxRulesGroup->active = $command->isEnabled();
         }
 
-        try {
-            if (false === $taxRulesGroup->validateFields(false)
-                || false === $taxRulesGroup->validateFieldsLang(false)) {
-                throw new TaxRulesGroupException('Tax Rules Group contains invalid field values');
-            }
-            if (false === $taxRulesGroup->update()) {
-                throw new CannotUpdateTaxRulesGroupException(
-                    sprintf(
-                        'Failed to update tax rules group with id %s',
-                        $command->getTaxRulesGroupId()->getValue()
-                    )
-                );
-            }
-            if (null !== $command->getShopAssociation()) {
-                $this->associateWithShops($taxRulesGroup, $command->getShopAssociation());
-            }
+        $shopIds = [];
+        foreach ($command->getShopAssociation() ?? [] as $shopId) {
+            $shopIds[] = new ShopId($shopId);
+        }
 
-            /* @phpstan-ignore-next-line */
-            if (!$taxRulesGroup->update()) {
-                throw new TaxRulesGroupException(sprintf('Cannot update tax rules group with id "%s"', $taxRulesGroup->id));
-            }
+        try {
```

Comment:
These try catches are not needed anymore, as the update/add method will already wrap it up and throw the right exception as defined in the repo here https://github.com/PrestaShop/PrestaShop/pull/28812/commits/97fd74c9556d07ee52d077e76656fdeca745d10a#diff-a8265060e5294d6b95986782f8f92bb0a91ad783463d155cec8cd12ee6e2ecabR152 (like CannotAddTaxRulesGroup or CannotUpdateTaxRulesGroup exceptopn)

Reply:
Fixed